### PR TITLE
ci: Add authors to release notes

### PR DESCRIPTION
**Pull request by mafredri**
This PR adds PR authors to release notes.

Currently co-authors are not supported, perhaps in a future enchancement.

```markdown
## Changelog

### Features

- b045734b6 feat: Add use template button to template row (#5811) ([BrunoQuaresma](https://github.com/BrunoQuaresma))
- bbb208e29 feat: Add CLI support for workspace build parameters (#5768) ([mtojek](https://github.com/mtojek))
- 443e2180f feat: indicate when workspace builds are stopped/started by Coder (#5813) ([Kira-Pilot](https://github.com/Kira-Pilot))

### Bug fixes

- 0e58772f5 fix: filter workspace_build resource on start/stop action (#5809) ([Kira-Pilot](https://github.com/Kira-Pilot))
- 37628c8b5 fix: comment on winget PR correctly (#5789) ([deansheather](https://github.com/deansheather))
- d2ae16dd2 fix: routinely ping agent websocket to ensure liveness (#5824) ([coadler](https://github.com/coadler))
- c3731a1be fix: ensure agent websocket only removes its own conn (#5828) ([coadler](https://github.com/coadler))
- dbfeb5630 fix: update docs link for configuration (#5827) ([kylecarbs](https://github.com/kylecarbs))

### Documentation

- f67acac2b docs: updated diagram with git provider logos (#5816) ([sharkymark](https://github.com/sharkymark))

### Tests

- 6a245ab1c test: Fix GPG test so it does not inherit parent parallelism (#5820) ([mafredri](https://github.com/mafredri))

### Continuous integration

- 8a9661e3c ci: Add authors to release notes

### Chores

- 918c37c35 chore: Ignore expired at on chromatic (#5808) ([BrunoQuaresma](https://github.com/BrunoQuaresma))
- 8afdf24d1 chore: Update sqlc to v1.16.0 (#5788) ([mafredri](https://github.com/mafredri))
- 73afdd7c0 chore: agent_test.go: use ptty.Peek() instead of expecting caret in TestAgent_SessionTTYShell (#5821) ([johnstcn](https://github.com/johnstcn))
- ba8dd496c chore: manually specify templates to embed (#5823) ([kylecarbs](https://github.com/kylecarbs))
- 882832cc5 chore: removed FE code owners (#5826) ([Kira-Pilot](https://github.com/Kira-Pilot))

Compare: [`v0.15.1...v0.15.2`](https://github.com/coder/coder/compare/v0.15.1...v0.15.2)

## Container image

- `docker pull ghcr.io/coder/coder:v0.15.2`
```

### CODE REVIEW

**Review by kylecarbs — APPROVED** [review]
I like that it makes working at Coder more like working on an OSS project. Neato!

File: scripts/release/generate_release_notes.sh [hunk 1]
```diff
@@ -112,7 +112,10 @@ for cat in "${!section_titles[@]}"; do
 done
 
 for commit in "${commits[@]}"; do
-	line="- $commit ${COMMIT_METADATA_TITLE[$commit]}\n"
+	line="- $commit ${COMMIT_METADATA_TITLE[$commit]}"
+	if [[ -v COMMIT_METADATA_AUTHORS[$commit] ]]; then
+		line+=" (${COMMIT_METADATA_AUTHORS[$commit]})"
```
Comment (deansheather) [inline scripts/release/generate_release_notes.sh:117]: Does the markdown link make them show up as a contributor in the release (this is a newish GH feature)? If not we need to use @ instead I think

**Review by deansheather — APPROVED** [review]

File: scripts/release/generate_release_notes.sh (hunk 1 above)
Comment (mafredri) [reply scripts/release/generate_release_notes.sh:117]: I have no idea, but I can def. revert back to @

File: scripts/release/check_commit_metadata.sh [hunk 2]
```diff
@@ -104,6 +108,9 @@ main() {
 		title=${parts[*]:2}
 		title=${title%$'\n'}
 		COMMIT_METADATA_TITLE[$commit_sha_short]=$title
+		if [[ -v authors[$commit_sha_long] ]]; then
+			COMMIT_METADATA_AUTHORS[$commit_sha_short]="[${authors[$commit_sha_long]}](https://github.com/${authors[$commit_sha_long]})"
```
Comment (mafredri) [inline scripts/release/check_commit_metadata.sh:112]:
```suggestion
			COMMIT_METADATA_AUTHORS[$commit_sha_short]="@${authors[$commit_sha_long]}"
```

**Pull request merged**